datalad/interface/download_url.py

```diff
@@ -35,77 +39,137 @@ class DownloadURL(Interface):
+                 archive=False, save=True, message=None):
         from ..downloaders.providers import Providers
 
+        try:
+            ds = require_dataset(
+                dataset, check_installed=True,
+                purpose='downloading urls')
+        except NoDatasetArgumentFound:
+            ds = None
+
+        common_report = {"action": "download_url",
+                         "ds": ds}
+
         urls = assure_list_from_str(urls)
 
-        if len(urls) > 1:
-            if path:
-                if not(isdir(path)):
-                    raise ValueError(
-                        "When specifying multiple urls, --path should point to "
-                        "an existing directory. Got %r" % path)
+        if len(urls) > 1 and path and not isdir(path):
+            yield get_status_dict(
+                status="error",
+                message=(
+                    "When specifying multiple urls, --path should point to "
+                    "an existing directory. Got %r", path),
+                type="file",
+                path=path,
+                **common_report)
+            return
         if not path:
-            path = curdir
+            path = ds.path if ds else curdir
 
         # TODO setup fancy ui.progressbars doing this in parallel and reporting overall progress
         # in % of urls which were already downloaded
         providers = Providers.from_config_files()
-        downloaded_paths, failed_urls = [], []
+        downloaded_paths = []
+        path_urls = {}
         for url in urls:
             # somewhat "ugly"
             # providers.get_provider(url).get_downloader(url).download(url, path=path)
             # for now -- via sugaring
             try:
                 downloaded_path = providers.download(url, path=path, overwrite=overwrite)
-                downloaded_paths.append(downloaded_path)
-                # ui.message("%s -> %s" % (url, downloaded_path))
             except Exception as e:
-                failed_urls.append(url)
-                ui.error(exc_str(e))
-                if stop_on_failure:
-                    break
-        if failed_urls:
-            raise RuntimeError("%d url(s) failed to download" % len(failed_urls))
-        return downloaded_paths
+                yield get_status_dict(
+                    status="error",
+                    message=exc_str(e),
+                    type="file",
+                    path=path,
+                    **common_report)
+            else:
+                downloaded_paths.append(downloaded_path)
+                path_urls[downloaded_path] = url
+                yield get_status_dict(
+                    status="ok",
+                    type="file",
+                    path=downloaded_path,
+                    **common_report)
+
+        if downloaded_paths and save and ds is not None:
+            msg = message or """\
+[DATALAD] Download URLs
+
+path: {}
```

Comment:
eh, good point.  When you give a subdirectory, it will be relative to the dataset, which I'd say isn't bad like a full path but still not very useful.  I'll drop the it.